

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>A PBGL Snakemake Workflow and Quatitative Trait Locus on Bulk Segregant Analysis &mdash; Oryza_Satvia_Snakemake_QTL 1.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="_static/doctools.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Welcome to Oryza_Satvia_Snakemake_QTLâ€™s documentation!" href="index.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> Oryza_Satvia_Snakemake_QTL
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">A PBGL Snakemake Workflow and Quatitative Trait Locus on Bulk Segregant Analysis</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#software-prerequisites">Software Prerequisites</a></li>
<li class="toctree-l2"><a class="reference internal" href="#painting-the-chromosomes-green">Painting The Chromosomes Green</a></li>
<li class="toctree-l2"><a class="reference internal" href="#high-impact-variants-in-qtl-region">HIGH IMPACT VARIANTS in QTL Region</a></li>
</ul>
</li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Oryza_Satvia_Snakemake_QTL</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>A PBGL Snakemake Workflow and Quatitative Trait Locus on Bulk Segregant Analysis</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/SMWFQTL.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <section id="a-pbgl-snakemake-workflow-and-quatitative-trait-locus-on-bulk-segregant-analysis">
<h1>A PBGL Snakemake Workflow and Quatitative Trait Locus on Bulk Segregant Analysis</h1>
<hr class="docutils" />
<section id="software-prerequisites">
<h2>Software Prerequisites</h2>
<p>#sra-toolkit
#Download git repository:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="c1">#Clone pbgl DNA Proto Workflow</span>
     git clone https://github.com/pbgl/dna-proto-workflow.git

     <span class="c1">#Change Directory to root or project</span>
     <span class="nb">cd</span> dna-proto-workflow

     <span class="c1">#Create virtual environment</span>
     Mamba env create --file env/all-dependencies.yml

     <span class="c1">#Activate the environment</span>
     conda activate dna-proto

     <span class="c1">#Make a new directory FASTQ</span>
     mkdir FASTQ

     <span class="c1">#Change directory to FASTQ</span>
     <span class="nb">cd</span> FASTQ

     <span class="c1">#Download data from NCBI</span>
     <span class="c1">#Start with Extremely Tolerant Rice Pool (&quot;Mutant&quot;)</span>
     wget https://sra-pub-run-odp.s3.amazonaws.com/sra/SRR834931/SRR834931

     <span class="c1">#Use sra-toolkit to split SRA into foward and reverse FASTQ reads and compress the files</span>
     fastq-dump --gzip --split-3

     <span class="c1">#Rename foward and reverse read files</span>
     ET_385_1.fq.gz
     ET_385_2.fq.gz

     <span class="c1">#Now get Extremely Sensitive Rice Pool (&quot;Wild-Type&quot;)</span>
     wget https://sra-pub-run-odp.s3.amazonaws.com/sra/SRR834927/SRR834927

     <span class="c1">#Use sra-toolkit to split SRA into foward and reverse FASTQ reads and compress the files</span>
     fastq-dump --gzip --split-3

     <span class="c1">#Rename foward and reverse read files</span>
     ES_430_1.fq.gz
     ES_430_2.fq.gz

     <span class="c1">#Follow technical documentation for the DNA Prototype Workflow</span>
     https://dna-proto-workflow-master.readthedocs.io/en/latest/

     <span class="c1">#Download Reference Genome and Index it</span>
     <span class="c1">#Download annotation files (gff, gtf, protein)</span>
     <span class="c1">#Build a snpEFF configuration file with soft links if necessary or preferred.</span>
     <span class="c1">#Provide Meta Data information (Contigs of interest, sample sets and definitions)</span>
     <span class="c1">#Configure config.yml and snpEff.config to match file pathways and names.\</span>

     <span class="c1">#Test your workflow on a dry-run</span>
     snakemake -npr
</pre></div>
</div>
<img alt="_images/screen1.png" src="_images/screen1.png" />
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>snakemake --dag -npr -j <span class="m">1</span> <span class="p">|</span> dot -Tsvg &gt; dag.svg
</pre></div>
</div>
<a class="reference internal image-reference" href="_images/dag.svg"><img alt="Directed Acrylic Graph" height="35800" src="_images/dag.svg" width="421280" /></a>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>snakemake -j <span class="m">4</span> -kpr

<span class="c1">#After PBGL Dna prototype pipeline root directory contains 634 items amounting to 271.4 Gigabytes</span>
<span class="c1">#There should be a vcf file in output variants final, lets take a look!</span>

bcftools view /home/michael/dna-proto-workflow/output/variants/final/freebayes~bwa~GCF_001433935.1_IRGSP-1.0~all_samples~filtered-strict.vcf.gz <span class="p">|</span> less -S

<span class="c1">#Filter variants for biallelic sites only and SNPS types</span>

bcftools view -m2 -M2 -v snps -o freebayes~bwa~GCF_001433935.1_IRGSP-1.0~all_samples~filtered-strict.Biallelic.vcf.gz freebayes~bwa~GCF_001433935.1_IRGSP-1.0~all_samples~filtered-strict.vcf.gz

<span class="c1">#Follow pbgl online documentation</span>

https://github.com/pbgl/QTLseqr


conda deactivate dna-proto


<span class="c1">#Create and activate new R Environment</span>

mamba env create --file env/R.yaml


<span class="c1">#Set samples</span>
HighBulk &lt;- <span class="s2">&quot;ET_385&quot;</span> <span class="c1"># SRA-run: SRR834931</span>
LowBulk &lt;- <span class="s2">&quot;ES_430&quot;</span> <span class="c1"># SRA-run: SRR834927</span>

<span class="c1">#set file name of the VCF file to load</span>
file &lt;- <span class="s2">&quot;freebayes~bwa~GCF_001433935.1_IRGSP-1.0~all_samples~filtered-strict~snpEff.Biallelic.vcf.gz&quot;</span>

<span class="c1">#Specify which chromosomes should be included in the analysis (i.e., exclude smaller contigs)</span>
Chroms &lt;- c<span class="o">(</span><span class="s2">&quot;NC_029256.1&quot;</span>,
<span class="s2">&quot;NC_029257.1&quot;</span>,
<span class="s2">&quot;NC_029258.1&quot;</span>,
<span class="s2">&quot;NC_029259.1&quot;</span>,
<span class="s2">&quot;NC_029260.1&quot;</span>,
<span class="s2">&quot;NC_029261.1&quot;</span>,
<span class="s2">&quot;NC_029262.1&quot;</span>,
<span class="s2">&quot;NC_029263.1&quot;</span>,
<span class="s2">&quot;NC_029264.1&quot;</span>,
<span class="s2">&quot;NC_029265.1&quot;</span>,
<span class="s2">&quot;NC_029266.1&quot;</span>,
<span class="s2">&quot;NC_029267.1&quot;</span><span class="o">)</span>
df &lt;-
importFromVCF<span class="o">(</span>
<span class="nv">file</span> <span class="o">=</span> file,
<span class="nv">highBulk</span> <span class="o">=</span> HighBulk,
<span class="nv">lowBulk</span> <span class="o">=</span> LowBulk,
<span class="nv">chromList</span> <span class="o">=</span> Chroms
<span class="o">)</span>


df_filt &lt;-
filterSNPs<span class="o">(</span>
<span class="nv">SNPset</span> <span class="o">=</span> df,
<span class="nv">refAlleleFreq</span> <span class="o">=</span> <span class="m">0</span>.20,
<span class="nv">minTotalDepth</span> <span class="o">=</span> <span class="m">70</span>,
<span class="nv">maxTotalDepth</span> <span class="o">=</span> <span class="m">200</span>,
<span class="nv">minSampleDepth</span> <span class="o">=</span> <span class="m">30</span>,
<span class="nv">depthDifference</span> <span class="o">=</span> <span class="m">100</span>,
<span class="c1">#minGQ = 150,</span>
<span class="nv">verbose</span> <span class="o">=</span> TRUE
<span class="o">)</span>

df_filt &lt;-
runGprimeAnalysis<span class="o">(</span>
<span class="nv">SNPset</span> <span class="o">=</span> df_filt,
<span class="nv">windowSize</span> <span class="o">=</span> 1e6,
<span class="nv">outlierFilter</span> <span class="o">=</span> <span class="s2">&quot;deltaSNP&quot;</span>,
<span class="nv">filterThreshold</span> <span class="o">=</span> <span class="m">0</span>.1
<span class="o">)</span>

df_filt &lt;-
runQTLseqAnalysis<span class="o">(</span>
<span class="nv">SNPset</span> <span class="o">=</span> df_filt,
<span class="nv">windowSize</span> <span class="o">=</span> 1e6,
<span class="nv">popStruc</span> <span class="o">=</span> <span class="s2">&quot;F2&quot;</span>,
<span class="nv">bulkSize</span> <span class="o">=</span> c<span class="o">(</span><span class="m">385</span>, <span class="m">430</span><span class="o">)</span>,
<span class="nv">replications</span> <span class="o">=</span> <span class="m">10000</span>,
<span class="nv">intervals</span> <span class="o">=</span> c<span class="o">(</span><span class="m">95</span>, <span class="m">99</span><span class="o">)</span>
<span class="o">)</span>

plotQTLStats<span class="o">(</span><span class="nv">SNPset</span> <span class="o">=</span> df_filt, <span class="nv">var</span> <span class="o">=</span> <span class="s2">&quot;nSNPs&quot;</span>, <span class="nv">plotIntervals</span> <span class="o">=</span> TRUE<span class="o">)</span>
</pre></div>
</div>
<figure class="align-default">
<img alt="_images/nsnps.png" src="_images/nsnps.png" />
</figure>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>plotQTLStats<span class="o">(</span><span class="nv">SNPset</span> <span class="o">=</span> df_filt, <span class="nv">var</span> <span class="o">=</span> <span class="s2">&quot;deltaSNP&quot;</span>, <span class="nv">plotIntervals</span> <span class="o">=</span> TRUE<span class="o">)</span>
</pre></div>
</div>
<figure class="align-default">
<img alt="_images/delta.png" src="_images/delta.png" />
</figure>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>plotQTLStats<span class="o">(</span><span class="nv">SNPset</span> <span class="o">=</span> df_filt, <span class="nv">var</span> <span class="o">=</span> <span class="s2">&quot;Gprime&quot;</span>, <span class="nv">plotThreshold</span> <span class="o">=</span> TRUE, <span class="nv">q</span> <span class="o">=</span> <span class="m">0</span>.02<span class="o">)</span>
</pre></div>
</div>
<figure class="align-default">
<img alt="_images/Gprime.png" src="_images/Gprime.png" />
</figure>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>df_filt<span class="nv">$p1</span> &lt;- df_filt<span class="nv">$AD_ALT</span>.LOW/df_filt<span class="nv">$DP</span>.LOW
df_filt<span class="nv">$p2</span> &lt;- df_filt<span class="nv">$AD_ALT</span>.HIGH/df_filt<span class="nv">$DP</span>.HIGH
df_filt3 &lt;- df_filt %&gt;% dplyr::filter<span class="o">(</span><span class="nv">CHROM</span><span class="o">==</span><span class="s2">&quot;NC_029263.1&quot;</span><span class="o">)</span>
<span class="c1">#Perhaps my filter is biased, but here it is. I want observed variants with pvalues below a significant threshold.</span>
df_filt4 &lt;- df_filt3 %&gt;% dplyr::filter<span class="o">(</span>pvalue &lt; .0000962<span class="o">)</span>
write.csv<span class="o">(</span>df_filt3,file <span class="o">=</span> <span class="s2">&quot;QTL.csv&quot;</span>,sep<span class="o">=</span><span class="s2">&quot;,&quot;</span><span class="o">)</span>
</pre></div>
</div>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>bash IGV.sh
</pre></div>
</div>
<a class="reference external image-reference" href="../images/IGV.png"><img alt="Integrated Genomic Viewer" src="_images/IGV.png" /></a>
</section>
<section id="painting-the-chromosomes-green">
<h2>Painting The Chromosomes Green</h2>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="c1"># Clone VCF Hunter from github</span>
git clone https://github.com/SouthGreenPlatform/VcfHunter.git
<span class="nb">cd</span> VcfHunter

<span class="c1"># Use vcf from Snakemake workflow as in input file to Filter it according to python script vcfFilter.1.0.py</span>
python vcfFilter.1.0.py --vcf freebayes~bwa~GCF_001433935.1_IRGSP-1.0~all_samples~filtered-strict~snpEff.vcf.gz --names Oryza_Satvia_Snakemake/PaintTheChromosomes/all_names.tab  --MinCov <span class="m">10</span> --MaxCov <span class="m">300</span> --MinAl <span class="m">3</span> --nMiss <span class="m">1</span> --RmAlAlt <span class="m">1</span>:3:4:5:6 --prefix DNAseq_Filtered -g y

<span class="c1">#Use vcftools to split vcf into independent vcf files per chromosome</span>
vcftools --gzvcf DNAseq_Filtered_filt.vcf.gz --chr NC_029256.1 --recode --out data/Chr01_DNAseq_Filtered.vcf.gz
vcftools --gzvcf DNAseq_Filtered_filt.vcf.gz --chr NC_029256.1 --recode --out data/Chr01_DNAseq_Filtered.vcf.gz
vcftools --gzvcf Downloads/VcfHunter/bin/DNAseq_Filtered_filt.vcf.gz --chr NC_029257.1 --recode --out data/Chr02_DNAseq_Filtered.vcf.gz
vcftools --gzvcf DNAseq_Filtered_filt.vcf.gz --chr NC_029258.1 --recode --out data/Chr03_DNAseq_Filtered.vcf.gz
vcftools --gzvcf DNAseq_Filtered_filt.vcf.gz --chr NC_029259.1 --recode --out data/Chr04_DNAseq_Filtered.vcf.gz
vcftools --gzvcf DNAseq_Filtered_filt.vcf.gz --chr NC_029260.1 --recode --out data/Chr05_DNAseq_Filtered.vcf.gz
vcftools --gzvcf DNAseq_Filtered_filt.vcf.gz --chr NC_029261.1 --recode --out data/Chr06_DNAseq_Filtered.vcf.gz
vcftools --gzvcf DNAseq_Filtered_filt.vcf.gz --chr NC_029262.1 --recode --out data/Chr07_DNAseq_Filtered.vcf.gz
vcftools --gzvcf DNAseq_Filtered_filt.vcf.gz --chr NC_029263.1 --recode --out data/Chr08_DNAseq_Filtered.vcf.gz
vcftools --gzvcf DNAseq_Filtered_filt.vcf.gz --chr NC_029264.1 --recode --out data/Chr09_DNAseq_Filtered.vcf.gz
vcftools --gzvcf DNAseq_Filtered_filt.vcf.gz --chr NC_029265.1 --recode --out data/Chr10_DNAseq_Filtered.vcf.gz
vcftools --gzvcf DNAseq_Filtered_filt.vcf.gz --chr NC_029266.1 --recode --out data/Chr11_DNAseq_Filtered.vcf.gz
vcftools --gzvcf DNAseq_Filtered_filt.vcf.gz --chr NC_029267.1 --recode --out data/Chr12_DNAseq_Filtered.vcf.gz

<span class="c1">#Use python script to produce an allelic frequency plot</span>
python vcf2allPropAndCovByChr.py --conf config/Rice.conf --origin config/RiceOrgin.tab --acc D2_F2_tt --ploidy <span class="m">2</span> --dcurve y --col /config/Color.conf --NoMiss --all y
python vcf2allPropAndCovByChr.py --conf config/Rice.conf --origin config/RiceOrigin.tab --acc ES_430,ET_385 --ploidy <span class="m">2</span> --NoMiss --all y


<span class="c1">#Allelic Frequency Plot (Tolerant Pool ET_385)</span>
</pre></div>
</div>
<img alt="_images/ET_385Ratio.png" src="_images/ET_385Ratio.png" />
</section>
<section id="high-impact-variants-in-qtl-region">
<h2>HIGH IMPACT VARIANTS in QTL Region</h2>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>Download:
SNPEff.latest
https://pcingola.github.io/SnpEff/download/

<span class="nb">cd</span> snpEff

cat freebayes~bwa~GCF_001433935.1_IRGSP-1.0~all_samples~filtered-strict~snpEff.vcf <span class="p">|</span> scripts/vcfAnnFirst.py <span class="p">|</span> java -jar SnpSift.jar extractFields - CHROM POS ID REF ALT QUAL FILTER <span class="s2">&quot;ANN[*].IMPACT&quot;</span> FORMAT ES_430 ET_385 &gt; impact.tsv

<span class="c1">#Import Data into R</span>

All_HIGH_IMPACT &lt;- impact %&gt;% dplyr::filter<span class="o">(</span>ANN....IMPACT <span class="o">==</span> <span class="s2">&quot;HIGH&quot;</span><span class="o">)</span>
df_filt &lt;- All_HIGH_IMPACT  %&gt;% dplyr::filter<span class="o">(</span><span class="nv">CHROM</span><span class="o">==</span><span class="s2">&quot;NC_029263.1&quot;</span><span class="o">)</span>
<span class="c1">#QTL 1 Region on Chromosome 8 from QTL.csv file</span>
df_filt &lt;- df_filt %&gt;% dplyr::filter<span class="o">(</span>POS &gt;<span class="o">=</span> <span class="m">17944918</span> <span class="p">&amp;</span> POS &lt;<span class="o">=</span> <span class="m">25698533</span><span class="o">)</span>
write.csv<span class="o">(</span>All_HIGH_IMPACT,file<span class="o">=</span><span class="s2">&quot;All_High_Impact.&quot;</span>,sep <span class="o">=</span> <span class="s2">&quot;/t&quot;</span><span class="o">)</span>
write.table<span class="o">(</span>All_HIGH_IMPACT,file <span class="o">=</span> <span class="s2">&quot;All_High_Impact.txt&quot;</span>,col.names <span class="o">=</span> FALSE<span class="o">)</span>

<span class="c1">#We know from QTL.csv Chromosome 6 &quot;NC_029263.1&quot; has a QTL peak in the range of positions 17944918 - 25698533!</span>
<span class="c1">#Filter original SNP set to include this range of positions</span>

df_filt3 &lt;- df_filt %&gt;% dplyr::filter<span class="o">(</span><span class="nv">CHROM</span><span class="o">==</span><span class="s2">&quot;NC_029263.1&quot;</span><span class="o">)</span>

<span class="c1">#QTL1 Region</span>
df_filt4 &lt;- df_filt3 %&gt;% dplyr::filter<span class="o">(</span>POS &gt;<span class="o">=</span> <span class="m">17944918</span> <span class="p">&amp;</span> POS &lt;<span class="o">=</span> <span class="m">25698533</span><span class="o">)</span>
<span class="c1">#Significant SNP</span>
df_filt4 &lt;- df_filt4 %&gt;% dplyr::filter<span class="o">(</span>pvalue &lt; .05<span class="o">)</span>
<span class="c1">#Write a text file</span>
write.table<span class="o">(</span>df_filt4, <span class="nv">file</span> <span class="o">=</span> <span class="s2">&quot;POSQTL1.txt&quot;</span>,col.names <span class="o">=</span> FALSE<span class="o">)</span>

<span class="c1">#Use MatchList.py script to find Variants that are</span>
nano MatchList.py
<span class="c1">#Make sure file names match what you wrote</span>


<span class="c1">#R Script</span>
<span class="c1">#df1 &lt;- data.frame(POS1=c(123,457,666,789))</span>
<span class="c1">#df2 &lt;- data.frame(POS2=c(123,444,566,789))</span>
<span class="c1">#write.table(df1,file=&quot;Variant1.txt&quot;,col.names = FALSE,row.names = FALSE)</span>
<span class="c1">#write.table(df2,file=&quot;Variant2.txt&quot;,col.names = FALSE,row.names = FALSE)</span>
import pandas as pd
with open<span class="o">(</span><span class="s2">&quot;POSQTL1.txt&quot;</span><span class="o">)</span> as file:
<span class="nv">list1</span> <span class="o">=</span> <span class="o">[]</span>
<span class="k">for</span> line <span class="k">in</span> file:
list1.append<span class="o">(</span>line.strip<span class="o">())</span>
<span class="c1">#print(list1)</span>
with open<span class="o">(</span><span class="s2">&quot;HighImpactQTL1.txt&quot;</span><span class="o">)</span> as file:
<span class="nv">list2</span> <span class="o">=</span> <span class="o">[]</span>
<span class="k">for</span> line <span class="k">in</span> file:
list2.append<span class="o">(</span>line.strip<span class="o">())</span>
<span class="c1">#print(list2)</span>
<span class="nv">found</span> <span class="o">=</span> <span class="o">[]</span>
<span class="k">for</span> i <span class="k">in</span> list1:
<span class="k">for</span> j <span class="k">in</span> list2:
<span class="k">if</span> j <span class="k">in</span> i:
    found.append<span class="o">(</span>j<span class="o">)</span>
<span class="c1">#print(found)</span>
<span class="c1">#print(type(found))</span>
<span class="nv">found</span><span class="o">=</span><span class="s1">&#39;,&#39;</span>.join<span class="o">(</span>found<span class="o">)</span>
<span class="c1">#print(found)</span>
<span class="c1">#print(type(found))</span>
<span class="nv">df</span> <span class="o">=</span> pd.DataFrame<span class="o">([</span>x.split<span class="o">(</span><span class="s1">&#39;,&#39;</span><span class="o">)</span> <span class="k">for</span> x <span class="k">in</span> found.split<span class="o">(</span><span class="s1">&#39;/n&#39;</span><span class="o">)])</span>
<span class="c1">#print(df.transpose())</span>
<span class="c1">#print(df)</span>
<span class="c1">#print(type(df))</span>
<span class="nv">dfT</span> <span class="o">=</span> df.T
<span class="nv">df9</span><span class="o">=</span>dfT.drop_duplicates<span class="o">()</span>
print<span class="o">(</span>df9<span class="o">)</span>
<span class="c1">#print(df9[&#39;[0]&#39;])</span>

<span class="c1">#Upload the document into R and filter results in decending order as seen in *orderedVariants.txt&quot;</span>
</pre></div>
</div>
</section>
</section>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
      
        <a href="index.html" class="btn btn-neutral float-left" title="Welcome to Oryza_Satvia_Snakemake_QTLâ€™s documentation!" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2022, Michael Hall

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>